src/netex_idf/calendars.rs

```diff
@@ -507,5 +536,50 @@ mod tests {
             );
             assert!(validity_pattern_iterator.next().is_none());
         }
+
+        #[test]
+        fn only_one_active_day() {
+            let validity_period = ValidityPeriod {
+                start_date: Date::from_ymd(2019, 6, 1),
+                end_date: Date::from_ymd(2019, 7, 31),
+            };
+            let weekday_pattern = vec![Weekday::Sat, Weekday::Sun].into_iter().collect();
+            let active_day = Date::from_ymd(2019, 7, 6);
+            let day_type_assignments = vec![ActiveDay(active_day)];
+            let mut validity_pattern_iterator = ValidityPatternIterator::new(
+                &validity_period,
+                &weekday_pattern,
+                Some(&day_type_assignments),
+            );
+            let date = validity_pattern_iterator.next().unwrap();
+            assert_eq!(Date::from_ymd(2019, 7, 6), date);
+            assert!(validity_pattern_iterator.next().is_none());
+        }
+
+        #[test]
+        fn inactive_day_before_operating_period() {
+            let validity_period = ValidityPeriod {
+                start_date: Date::from_ymd(2019, 6, 1),
+                end_date: Date::from_ymd(2019, 7, 31),
+            };
+            let weekday_pattern = vec![Weekday::Sat, Weekday::Sun].into_iter().collect();
+            let operating_period = ValidityPeriod {
+                start_date: Date::from_ymd(2019, 7, 1),
```

Comment:
How about moving the start date to the 6th. This test is really about testing the inactive day, not the week pattern.
```suggestion
                start_date: Date::from_ymd(2019, 7, 6),
```

Reply:
I can change it but I "test" implicitly the weekly pattern as well.

Reply:
It's a unit test. Other tests already are taking care of testing the week pattern, that's why I mentioned it.